### PR TITLE
Features/fixpublishvsix

### LINKED ISSUE #13: Publish vsix doesn't apply "Tag"
The publish vsix task doesn't apply the "Tag" attribute while Publish from manifest and package from manifest do.

### PR DESCRIPTION
**Pull request by jessehouwing**
Fixes #13: Tag is not applied when publishing from VSIX
Adds: Support for Paid Extensions

### CODE REVIEW

File: BuildTasks/Common/Common.ts [hunk 1]
```diff
@@ -79,7 +79,25 @@ export function setTfxManifestArguments(tfx: ToolRunner): (() => void) {
         const isPreview = extensionVisibility.indexOf("preview") >= 0;
 
         jsonOverrides.public = isPublic;
-        if (isPreview) { jsonOverrides.galleryFlags = ["Preview"]; };
+
+        if (isPreview) {
+            jsonOverrides.galleryFlags = jsonOverrides.galleryFlags || [];
```
Comment (trevorsg) [inline BuildTasks/Common/Common.ts]: Looks like this line isn't doing anything

File: BuildTasks/PublishExtension/vsixeditor.ts [hunk 2]
```diff
@@ -134,12 +137,29 @@ export class VSIXEditor {
 
                 vsixmanifest.PackageManifest.Metadata.GalleryFlags = flagsEditor.toString();
             }
+            if (this.extensionPricing && this.extensionPricing !== "default") {
+                let flagsEditor = new GalleryFlagsEditor(vsixmanifest.PackageManifest.Metadata.GalleryFlags);
```
Comment (trevorsg) [inline BuildTasks/PublishExtension/vsixeditor.ts]: const?

Comment (jessehouwing) [inline BuildTasks/PublishExtension/vsixeditor.ts]: Yups.

File: BuildTasks/Common/Common.ts (hunk 1 above)
Comment (jessehouwing) [inline BuildTasks/Common/Common.ts]: Agreed. Replaced the line below with .Push. 
Kept the coalesce in there to make the code aligned with the pricing option below and to make sure it keeps working if the order is changed.

**Pull request merged**